src/goto-programs/string_instrumentation.cpp

```diff
@@ -841,10 +841,15 @@ void string_instrumentationt::invalidate_buffer(
     from_integer(0, cntr_sym.type),
     target->source_location));
 
-  goto_programt::targett check=dest.add_instruction();
-
-  goto_programt::targett invalidate=dest.add_instruction(ASSIGN);
-  invalidate->source_location=target->source_location;
+  // this instruction is created below
+  goto_programt::targett check = dest.add(goto_programt::instructiont());
```

Comment:
Shouldn't the construction be moved down so that a 'complete' constructor can be used? (several occurrences)

Reply:
done!